authorization/storage.go

```diff
@@ -2,33 +2,313 @@ package authorization
 
 import (
 	"context"
+	goerrors "errors"
+	"fmt"
+	"slices"
 
+	"github.com/go-crypt/crypt"
+	"github.com/influxdata/influxdb/v2"
 	"github.com/influxdata/influxdb/v2/kit/platform"
 	"github.com/influxdata/influxdb/v2/kit/platform/errors"
 	"github.com/influxdata/influxdb/v2/kit/tracing"
 	"github.com/influxdata/influxdb/v2/kv"
+	influxdb2_algo "github.com/influxdata/influxdb/v2/pkg/crypt/algorithm/influxdb2"
 	"github.com/influxdata/influxdb/v2/snowflake"
+	"go.uber.org/zap"
 )
 
+/*---
+Token storage and verification
```

Comment:
Excellent comment. I'm assuming that instructions from here and information will be ported in to our docs as well?

Reply:
Yes, the docs team should be able to use this for user-facing documentation.